### DISCUSSION TITLE
MemoryError Upon Reading in NumPy Array

### DISCUSSION THREAD

**Discussion by wshn25** [ARM-DOE/pyart#1057] [Q&A] [answered]
---------------------------------------------------------------------------
MemoryError                               Traceback (most recent call last)
/tmp/ipykernel_47938/1785810078.py in <module>
      1 new_time = np.array(time, dtype='datetime64[ns]')
----> 2 new_ref = np.array(DBZH, dtype='uint8')
      3 new_velv = np.array(VELV, dtype='uint8')
      4 new_velh = np.array(VELH, dtype='uint8')
      5 

MemoryError: Unable to allocate 712. MiB for an array with shape (288, 2592000) and data type uint8
____

That's the problem I faced. Here's the script I used:
new_time = np.array(time, dtype='datetime64[ns]')
new_ref = np.array(DBZH, dtype='uint8')
new_velv = np.array(VELV, dtype='uint8')
new_velh = np.array(VELH, dtype='uint8')
 _____

Can anyone help me to solve this issue? 
I have already changed dtype from float to uint8, and it seemed to get troubled.

**Comment by zssherman** [comment, marked as answer]
I'm not sure how your generating DBZH based on the code, but that array does seem rather large. I would try reading in those arrays with dask in chunks.

**Comment by wshn25** [comment]
Thank you for the help. I am working on QVP by using S-Band radar data. So, it seems difficult for me to define each wind component.
And it seems for another parameter also gets the same error with DBZH.
